### PR TITLE
Refactor releasing process

### PR DESCRIPTION
**Pull request by Felixoid**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Migrate package building to nfpm
- Deprecate `release` script in favor of `packages/build`
- Build everything in clickhouse/binary-builder image (cleanup: clickhouse/deb-builder)
- Add symbol stripping to cmake (todo: use $prefix/lib/$bin_dir/clickhouse/$binary.debug)
- Fix issue with DWARF symbols
- Add Alpine APK packages
- Rename `alien` to `additional_pkgs`

### CODE REVIEW

**Comment by Felixoid** [comment]
Need to add the new commit like https://github.com/ClickHouse/ClickHouse/pull/32218

**Comment by alesapin** [comment]
Here: https://github.com/ClickHouse/ClickHouse/pull/35196

File: docker/packager/packager [hunk 1]
```diff
@@ -77,46 +95,69 @@ def parse_env_variables(build_type, compiler, sanitizer, package_type, image_typ
...
+        cmake_flags.append("-DCMAKE_AUTOGEN_VERBOSE=ON")
+        cmake_flags.append("-DCMAKE_INSTALL_PREFIX=/usr")
+        cmake_flags.append("-DCMAKE_INSTALL_SYSCONFDIR=/etc")
+        cmake_flags.append("-DCMAKE_INSTALL_LOCALSTATEDIR=/var")
+        cmake_flags.append("-DINSTALL_STRIPPED_BINARIES=ON")
+
+    result.append(f"CC={cc}")
+    result.append(f"CXX={cxx}")
+    cmake_flags.append(f"-DCMAKE_C_COMPILER=`which {cc}`")
+    cmake_flags.append(f"-DCMAKE_CXX_COMPILER=`which {cxx}`")
```
Comment (Felixoid) [inline docker/packager/packager:160]: It doesn't work, `which` is executed on the host, not in the container.

**Comment by alesapin** [comment]
Related https://github.com/ClickHouse/ClickHouse/issues/35405

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
Great!

**Comment by makeavish** [comment]
Thanks for merging!
Will now ARM64 images be published on https://hub.docker.com/r/clickhouse/clickhouse-server/tags?

**Comment by Felixoid** [comment]
Hi @makeavish, subscribe on #35211

**Comment by guoxiaolongzte** [comment]
docker exec -lt  clickhouse/binary-builder:latest   
./build --rpm          


root@ad569b6587a0:~/ClickHouse/packages# pwd
/root/ClickHouse/packages
root@ad569b6587a0:~/ClickHouse/packages# ./build --rpm

Current version is 23.6.1.873
Building rpm package for clickhouse-client.yaml
using rpm packager...
matching "./root/usr/bin/clickhouse-compressor": file does not exist


### What error is this? I haven't seen any official website introduction

**Comment by Felixoid** [comment]
See https://github.com/ClickHouse/ClickHouse/pull/51679. The process of building in docker is documented there

**Pull request merged**